tests/test_resource_base.py

```diff
@@ -32,3 +32,47 @@ def test_validate_resource_id_custom_message(client):
         resource.validate_resource_id("invalid", message="No no no, only tests allowed here!")
 
     assert str(excinfo.value) == "No no no, only tests allowed here!"
+
+
+def test_response_handling_error_sets_message_and_key_correctly():
+    try:
+        raise ResponseHandlingError("This is a ResponseHandlingError", "test_idempotency_key")
+    except ResponseHandlingError as exc:
+        assert exc.idempotency_key == "test_idempotency_key"
+        assert exc.args[0] == "This is a ResponseHandlingError"
+
+
+def test_bad_request_error_set_message_and_key_correctly():
+    try:
+        raise BadRequestError({"detail": "This is a BadRequestError", "status": 400}, "test_idempotency_key")
+    except BadRequestError as exc:
+        assert exc.idempotency_key == "test_idempotency_key"
+        assert exc.args[0] == "This is a BadRequestError"
+
+
+def test_conflict_error_set_message_and_key_correctly():
+    try:
+        raise BadRequestError({"detail": "This is a ConflictError", "status": 409}, "test_idempotency_key")
+    except BadRequestError as exc:
+        assert exc.idempotency_key == "test_idempotency_key"
+        assert exc.args[0] == "This is a ConflictError"
+
+
+def test_create_request_without_idempotency_key_sets_idempotency_key_on_error(client, response):
+    """Test that an exception contains an imdepotency_key eventhough no idempotency_key was passed."""
```

Comment:
Maybe use the term `auto-generated` here (also for the test below), and fix the typo in `imdepotency_key` ;)